### PR TITLE
Introducing github username validation when updating student data and…

### PR DESCRIPTION
**Pull request by mae1579**
Introducing github username validation when updating student data and fix problem about one of student validation.

Signed-off-by: Mateusz Umięcki <mateuszumiecki@gmail.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,9 +22,11 @@
     "express-fileupload": "^1.4.0",
     "express-rate-limit": "^6.4.0",
     "firebase-functions": "^3.22.0",
+    "gh-account-exists": "^0.1.0",
```
Comment (patrycja-dev41171) [inline package.json:25]: It shouldn't be here, should it?

File: records/student.record.ts [hunk 2]
```diff
@@ -103,7 +103,7 @@ export class StudentRecord implements StudentEntity {
     if (obj.status !== undefined && obj.status !== null && !Object.values(Status).includes(obj.status)) {
       throw new ValidationError('Status has to be one of the enum Status options');
     }
-    if (obj.monthsOfCommercialExp !== undefined && obj.monthsOfCommercialExp !== null && obj.monthsOfCommercialExp > 0) {
+    if (obj.monthsOfCommercialExp !== undefined && obj.monthsOfCommercialExp !== null && obj.monthsOfCommercialExp < 0) {
       throw new ValidationError('MonthsOfCommercialExperience  must be greater than 0.');
     }
```
Comment (MiroslawChmielecki) [inline records/student.record.ts:108]: Very sharp eye necessary in IT ;)

File: records/student.record.ts [hunk 3]
```diff
@@ -103,7 +103,7 @@ export class StudentRecord implements StudentEntity {
     if (obj.status !== undefined && obj.status !== null && !Object.values(Status).includes(obj.status)) {
       throw new ValidationError('Status has to be one of the enum Status options');
     }
-    if (obj.monthsOfCommercialExp !== undefined && obj.monthsOfCommercialExp !== null && obj.monthsOfCommercialExp > 0) {
+    if (obj.monthsOfCommercialExp !== undefined && obj.monthsOfCommercialExp !== null && obj.monthsOfCommercialExp < 0) {
```
Comment (patrycja-dev41171) [inline records/student.record.ts:106]: A small error. It's good that you noticed him!

File: routers/studentBackData.ts [hunk 4]
```diff
@@ -75,23 +70,27 @@ studentBackRouter.post('/', async (req, res) => {
...
-    res.json('Zaktualizowano dane ');
-  } catch (err) {
-    console.log(err);
+    res.status(200).json('Zaktualizowano dane ');
+  } else {
+    throw new ValidationError(
+      `Użytkownik o takim  loginie:  ${studentData.githubUserName}  nie istnieje na githubie ! ,sprawdz login i spróbuj ponownie wpisać poprawny login z github. `
+    );
   }
 });
```
Comment (MiroslawChmielecki) [inline routers/studentBackData.ts:96]: Perfect, I need this message in Front-End ;)

**Review by MiroslawChmielecki — APPROVED** [review]

File: routers/studentBackData.ts [hunk 5]
```diff
@@ -75,23 +70,27 @@ studentBackRouter.post('/', async (req, res) => {
...
-    }
+  const studentDb = await StudentRecord.getOneByEmail(email);
+  // if(studentDb.firstName)
+  if (!studentDb) {
+    throw new ValidationError(
+      'Nie możesz zaktualizować danych ponieważ twoje konto z twoim emailem nie jest wpisane do naszej bazy danych, skontaktuj się z administracją w celu dalszej pomocy.'
+    );
+  }
+  const response = await fetch(`https://api.github.com/users/${githubUserName}`);
+  const data = await response.json();
```
Comment (patrycja-dev41171) [inline routers/studentBackData.ts:81]: It could have been locked in try {} catch () {}.

**Review by patrycja-dev41171 — COMMENTED** [review]
Very good, but most of all validation and data checking will help the GH API.

**Pull request merged**